atkinson/dlrn/http_data.py

```diff
@@ -69,81 +77,97 @@ def dlrn_http_factory(host, config_file=None, logger=getLogger()):
     if host not in config:
         return None
 
+    link = config[host]['link']
+    if link_name is not None:
+        link = link_name
+
     return DlrnHttpData(config[host]['url'],
                         config[host]['release'],
+                        link_name=link,
                         logger=logger)
 
 
 class DlrnHttpData():
     """A class used to interact with the dlrn API"""
-    def __init__(self, url, release, logger=getLogger()):
+    def __init__(self, url, release, link_name='current', logger=getLogger()):
         """
         Class constructor
 
         :param url: The URL to the host to obtain data.
         :param releases: The release name to use for lookup.
+        :param link_name: The name of the dlrn symlink to fetch data from.
         :param logger: An atkinson logger to use. Default is the base logger.
         """
-        self.url = url
+        self.url = os.path.join(url, release)
         self.release = release
         self._logger = logger
+        self._link_name = link_name
+        self._commit_data = {}
+        self._fetch_commit()
+
+    def _fetch_commit(self):
+        """
+        Fetch the commit data from dlrn
+        """
+        full_url = os.path.join(self.url,
+                                self._link_name,
+                                'commit.yaml')
+        data = _fetch_yaml(full_url, self._logger)
+        if data is not None and 'commits' in data:
+            pkg = data['commits'][0]
+            if pkg['status'] == 'SUCCESS':
+                self._commit_data = {'name': pkg['project_name'],
+                                     'dist_hash': pkg['distro_hash'],
+                                     'commit_hash': pkg['commit_hash']}
+                if 'extended_hash' in pkg:
```

Comment:
You could lose this block if you wanted, using the dict.get() method, so it would collapse down to:
```
self._commit_data['extended_hash'] = pkg.get('extended_hash')
```
This will return None if you do not specify another default.  The only other change this would require is below, which I will give a separate comment to.